internal/applicationsnapshot/input.go

```diff
@@ -176,3 +210,79 @@ func readSnapshotSource(input []byte) (app.SnapshotSpec, error) {
 	log.Debugf("Read application snapshot from file %s", input)
 	return file, nil
 }
+
+func (*remoteClient) Get(ref name.Reference) (*remote.Descriptor, error) {
+	desc, err := remote.Get(ref)
+	if err != nil {
+		return nil, err
+	}
+	return desc, nil
+}
+
+func (*remoteClient) Index(ref name.Reference) (v1.ImageIndex, error) {
+	desc, err := remote.Index(ref)
+	if err != nil {
+		return nil, err
+	}
+	return desc, nil
+}
+
+func clientFromContext(ctx context.Context) (client, bool) {
+	cli, ok := ctx.Value(RemoteClientKey{}).(client)
+	return cli, ok
+}
+
+func expandImageIndex(ctx context.Context, snap *app.SnapshotSpec) error {
+	client, ok := clientFromContext(ctx)
+	if !ok {
+		client = &remoteClient{}
+	}
+	// For an image index, remove the original component and replace it with an expanded component with all its image manifests
+	var removeIndices []int
+	var modifiedComponents []app.SnapshotComponent
+	log.Debugf("Contents of snapComponents are %v", snap.Components)
```

Comment:
I believe `%#v` should display a more readable representation of the components.